### PR TITLE
Vulnerability Detector integration tests for Amazon Linux 2023

### PR DESCRIPTION
**Pull request by Leoquicenoz**
|Related issue|
|---|
|https://github.com/wazuh/wazuh-qa/issues/4440 |

## Description

For Wazuh v4.7.0 we want to add Amazon Linux 2023 OS support for the Wazuh Vulnerability Detector module.

The following test was added to cover the new supported OS version:

- `Feed tests:` These tests have to cover the addition of the new feed for `Amazon Linux 2023`
- `Provider tests`: Here we test the settings for `Amazon Linux` with OS version `2023`
- `Scan results tests`: Here we  add the tests that ensure `Amazon Linux 2023` agents are scanned properly

Modules Involved: 

- test_vulnerability_detector/test_feeds/test_download_feeds.py
- test_vulnerability_detector/test_feeds/test_duplicate_feeds.py
- test_vulnerability_detector/test_feeds/test_import_invalid_feed_type.py
- test_vulnerability_detector/test_feeds/test_validate_feed_content.py
- test_vulnerability_detector/test_providers/test_enabled.py
- test_vulnerability_detector/test_providers/test_missing_os.py
- test_vulnerability_detector/test_providers/test_os.py
- test_vulnerability_detector/test_providers/test_update_interval.py
- test_vulnerability_detector/test_scan_results/test_scan_provider_and_nvd_vulnerabilities.py
- test_vulnerability_detector/test_scan_results/test_scan_provider_vulnerabilities.py
- test_vulnerability_detector/test_scan_results/test_scan_vulnerability_removal.py

## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/| Ubuntu - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8764512/R3.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8764510/R1.zip)  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8764511/R2.zip)  | 2023/09/01 | @Leoquicenoz   | 

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

Important: Don't remove these checks if your PR modifies Python code.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 '''
-copyright: Copyright (C) 2015-2022, Wazuh Inc.
+copyright: Copyright (C) 2015-2023, Wazuh Inc.
```
Comment (MarcelKemp) [inline tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py:2]: Apply the following change to all modified copyrights, please.
```suggestion
copyright: Copyright (C) 2015, Wazuh Inc.
```

File: tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py [hunk 2]
```diff
@@ -96,7 +97,7 @@ def test_download_feeds(configuration, metadata, set_wazuh_configuration_vdt, tr
         - Check in log that the database provider has been updated successfully.
         - Check that the timestamp of the feed metadata does not exceed the established threshold limit.
 
-    wazuh_min_version: 4.4.0
+    wazuh_min_version: 4.7.0
```
Comment (MarcelKemp) [inline tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py:100]: Let's avoid changing the version.

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml [hunk 3]
```diff
@@ -8,6 +8,16 @@
     provider_name: Amazon Linux 2022
     update_interval: 5s
 
+- name: Amazon Linux 2023
```
Comment (MarcelKemp) [inline tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml:11]: For this case, we are going to overwrite the previous `amazon-linux-2022` with `amazon-linux-2023`.

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_no_agent_data.yaml [hunk 4]
```diff
@@ -33,6 +33,14 @@
...
     system: ALAS2
 
+
+- name: ALAS 2023
+  description: Scan ALAS vulnerabilities using only the NVD feed
+  configuration_parameters:
+    NVD_JSON_PATH: CUSTOM_NVD_JSON_PATH
+  metadata:
+    system: ALAS_2023
+
```
Comment (MarcelKemp) [inline tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_no_agent_data.yaml:43]: For this case, let's leave only the ALAS2 case, or else overwrite it by ALAS2023.

**Review by MarcelKemp — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_no_agent_data.yaml (hunk 4 above)
Comment (Leoquicenoz) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_no_agent_data.yaml]: Done in commit: refactor(#4440): Reduce the number of ALAS cases to one in some tests

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml (hunk 3 above)
Comment (Leoquicenoz) [reply tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml]: Done in commit: refactor(#4440): Reduce the number of ALAS cases to one in some tests

File: tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py (hunk 1 above)
Comment (Leoquicenoz) [reply tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py]: Done in commit: docs: Undo changes in wazuh_min_version and modify Copyright in some tests

File: tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py (hunk 2 above)
Comment (Leoquicenoz) [reply tests/integration/test_vulnerability_detector/test_feeds/test_download_feeds.py]: Done in commit: docs: Undo changes in wazuh_min_version and modify Copyright in some tests

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_updated_package_still_vulnerable.py
Comment (MarcelKemp) [inline tests/integration/test_vulnerability_detector/test_scan_results/test_scan_updated_package_still_vulnerable.py:1]: This file does not belong to `4.7.0`, but to `master` (_4.8.0_), so it does not need to be added to this branch.

File: tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_validate_json_feed_content.yaml [hunk 5]
```diff
@@ -39,6 +39,19 @@
...
+  configuration_parameters:
+  metadata:
+    provider_name: Amazon Linux 2023
+    expected_format: application/gzip
+    path: /tmp/alas.json.gz
+    extension: gz
+    decompressed_file: /tmp/alas.json
+    url: https://feed.wazuh.com/vulnerability-detector/ALAS/2023/alas2023.json.gz
+
+
```
Comment (MarcelKemp) [inline tests/integration/test_vulnerability_detector/test_feeds/data/test_cases/cases_validate_json_feed_content.yaml:54]: To fix this check:
- https://github.com/wazuh/wazuh-qa/actions/runs/6241022523/job/16942353629?pr=4482

Avoid adding the following line breaks:
```suggestion
- name: Amazon Linux 2023
  description: JSON Amazon Linux provider
  configuration_parameters:
  metadata:
    provider_name: Amazon Linux 2023
    expected_format: application/gzip
    path: /tmp/alas.json.gz
    extension: gz
    decompressed_file: /tmp/alas.json
    url: https://feed.wazuh.com/vulnerability-detector/ALAS/2023/alas2023.json.gz

```

**Review by MarcelKemp — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_updated_package_still_vulnerable.py
Comment (Leoquicenoz) [reply tests/integration/test_vulnerability_detector/test_scan_results/test_scan_updated_package_still_vulnerable.py:1]: Done in commit: refactor(#4440): Delete test_scan_updated_package_still_vulnerable.py file

**Review by MarcelKemp — APPROVED** [review]

**Pull request merged**